cmd/ui/main.go

```diff
@@ -1,50 +1,43 @@
 package main
 
 import (
+	"context"
 	"embed"
-	"encoding/json"
 	"io/fs"
 	"net/http"
 	"os"
 	"path/filepath"
 
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/sirupsen/logrus"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/applications"
+	"github.com/weaveworks/weave-gitops/pkg/server"
 )
 
+var log = logrus.New()
+
 func main() {
-	log := logrus.New()
 	mux := http.NewServeMux()
 
 	mux.Handle("/health/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-	}))
-
-	mux.Handle("/api/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		res := struct {
-			Ok bool `json:"ok"`
-		}{
-			Ok: true,
-		}
-
-		b, err := json.Marshal(res)
+		_, err := w.Write([]byte("ok"))
 
 		if err != nil {
-			log.Errorf("could not marshal: %s", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		if _, err := w.Write(b); err != nil {
-			log.Errorf("error writing bytes: %s", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+			log.Errorf("error writing health check: %s", err)
 		}
 	}))
 
 	assetFS := getAssets()
 	assetHandler := http.FileServer(http.FS(assetFS))
 	redirector := createRedirector(assetFS, log)
 
+	gMux := runtime.NewServeMux()
+	mux.Handle("/v1/", gMux)
+
+	if err := pb.RegisterApplicationsHandlerServer(context.Background(), gMux, server.NewApplicationsServer()); err != nil {
+		log.Fatalf("could not register application: %s", err)
+	}
```

Comment:
⭐ . Sorry I didn't understand the use case properly first. This is neat though being able to mux in the gateway here.

For things like metrics collection etc the proxy model might work better though? Then all api usage reporting is coming from the same container etc.

Reply:
Hmm yeah good point. The first use case I was optimizing for was the stand-alone binary running outside the cluster. Running inside the cluster will require some thought.